Cloud9 Issue

Make sure your running an up to date version of Hassbian before reporting a problem. If not make sure to specify what version and why so we can recreate the issue in question.

You should only file an issue if you found a bug. Feature and enhancement requests should go in #33 that's dedicated for this usage.

Hassbian-config version (hassbian-config -V):
0.11.1
Script/Suite:
sudo hassbian-config install cloud9
Description of problem:
Service will not start after installing.
Expected:

Traceback/log (if applicable):

pi@hassbian:~ $ sudo systemctl status cloud9@homeassistant.service
● cloud9@homeassistant.service - Cloud9
   Loaded: loaded (/etc/systemd/system/cloud9@homeassistant.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-12-05 07:46:02 EST; 1min 7s ago
  Process: 6032 ExecStart=/opt/c9sdk/server.js -l 0.0.0.0 -a : -w /home/homeassistant/c9workspace (code=exited, status=1/FAILURE)
 Main PID: 6032 (code=exited, status=1/FAILURE)

Dec 05 07:46:02 hassbian server.js[6032]:     at Module.require (internal/modules/cjs/loader.js:636:17)
Dec 05 07:46:02 hassbian server.js[6032]:     at require (internal/modules/cjs/helpers.js:20:18)
Dec 05 07:46:02 hassbian server.js[6032]:     at main (/opt/c9sdk/server.js:74:20)
Dec 05 07:46:02 hassbian server.js[6032]:     at Object.<anonymous> (/opt/c9sdk/server.js:55:5)
Dec 05 07:46:02 hassbian server.js[6032]:     at Module._compile (internal/modules/cjs/loader.js:688:30)
Dec 05 07:46:02 hassbian server.js[6032]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
Dec 05 07:46:02 hassbian server.js[6032]:     at Module.load (internal/modules/cjs/loader.js:598:32)
Dec 05 07:46:02 hassbian systemd[1]: cloud9@homeassistant.service: Main process exited, code=exited, status=1/FAILURE
Dec 05 07:46:02 hassbian systemd[1]: cloud9@homeassistant.service: Unit entered failed state.
Dec 05 07:46:02 hassbian systemd[1]: cloud9@homeassistant.service: Failed with result 'exit-code'.

Hi,
just tried now and still dosn't work.
I tried:

<pre>
sudo hassbian-config install cloud9 --force
</pre>

and here is the output:

<pre>
Cloud9 script for Hassbian.
This script was originally contributed by Ludeeus <https://github.com/ludeeus>.

Creating installation directory...
Changing to the homeassistant user
Downloading and installing Cloud9 SDK...
Cloning into '/opt/c9sdk'...
remote: Enumerating objects: 52812, done.
remote: Total 52812 (delta 0), reused 0 (delta 0), pack-reused 52812
Receiving objects: 100% (52812/52812), 35.30 MiB | 561.00 KiB/s, done.
Resolving deltas: 100% (32452/32452), done.
From https://github.com/c9/core
 * [new branch]        c9         -> c9/c9
 * [new branch]        gh-pages   -> c9/gh-pages
 * [new branch]        master     -> c9/master
 * [new branch]        smf-sdk    -> c9/smf-sdk
Already up-to-date.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11769  100 11769    0     0  17673      0 --:--:-- --:--:-- --:--:-- 17697
Installing base packages. Use --help for more options
:Installing Node v6.3.1
--2019-02-01 17:45:18--  https://nodejs.org/dist/v6.3.1/node-v6.3.1-linux-armv7l.tar.gz
Resolving nodejs.org (nodejs.org)... 104.20.23.46, 104.20.22.46, 2606:4700:10::6814:162e, ...
Connecting to nodejs.org (nodejs.org)|104.20.23.46|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12716600 (12M) [application/gzip]
Saving to: ‘node.tar.gz’

     0K .......... .......... .......... .......... ..........  0%  545K 23s
    50K .......... .......... .......... .......... ..........  0%  430K 26s
   100K .......... .......... .......... .......... ..........  1%  450K 26s
   150K .......... .......... .......... .......... ..........  1%  807K 23s
   200K .......... .......... .......... .......... ..........  2%  563K 23s
   250K .......... .......... .......... .......... ..........  2%  807K 21s
   300K .......... .......... .......... .......... ..........  2%  524K 22s
   350K .......... .......... .......... .......... ..........  3%  300K 24s
   400K .......... .......... .......... .......... ..........  3%  411K 24s
   450K .......... .......... .......... .......... ..........  4%  260K 26s
   500K .......... .......... .......... .......... ..........  4%  157K 31s
   550K .......... .......... .......... .......... ..........  4%  272K 32s
   600K .......... .......... .......... .......... ..........  5%  190K 34s
   650K .......... .......... .......... .......... ..........  5%  384K 34s
   700K .......... .......... .......... .......... ..........  6%  737K 32s
   750K .......... .......... .......... .......... ..........  6%  618K 31s
   800K .......... .......... .......... .......... ..........  6%  721K 30s
   850K .......... .......... .......... .......... ..........  7%  816K 29s
   900K .......... .......... .......... .......... ..........  7%  590K 29s
   950K .......... .......... .......... .......... ..........  8%  665K 28s
  1000K .......... .......... .......... .......... ..........  8%  683K 27s
  1050K .......... .......... .......... .......... ..........  8%  534K 27s
  1100K .......... .......... .......... .......... ..........  9%  814K 26s
  1150K .......... .......... .......... .......... ..........  9%  537K 26s
  1200K .......... .......... .......... .......... .......... 10%  544K 26s
  1250K .......... .......... .......... .......... .......... 10% 1.15M 25s
  1300K .......... .......... .......... .......... .......... 10%  698K 24s
  1350K .......... .......... .......... .......... .......... 11%  178K 26s
  1400K .......... .......... .......... .......... .......... 11%  789K 25s
  1450K .......... .......... .......... .......... .......... 12%  326K 25s
  1500K .......... .......... .......... .......... .......... 12%  460K 25s
  1550K .......... .......... .......... .......... .......... 12%  234K 26s
  1600K .......... .......... .......... .......... .......... 13%  237K 26s
  1650K .......... .......... .......... .......... .......... 13%  228K 27s
  1700K .......... .......... .......... .......... .......... 14%  358K 27s
  1750K .......... .......... .......... .......... .......... 14%  564K 26s
  1800K .......... .......... .......... .......... .......... 14%  696K 26s
  1850K .......... .......... .......... .......... .......... 15%  475K 26s
  1900K .......... .......... .......... .......... .......... 15%  501K 25s
  1950K .......... .......... .......... .......... .......... 16%  272K 26s
  2000K .......... .......... .......... .......... .......... 16%  211K 26s
  2050K .......... .......... .......... .......... .......... 16% 1.17M 25s
  2100K .......... .......... .......... .......... .......... 17%  331K 26s
  2150K .......... .......... .......... .......... .......... 17%  300K 26s
  2200K .......... .......... .......... .......... .......... 18%  163K 26s
  2250K .......... .......... .......... .......... .......... 18%  396K 26s
  2300K .......... .......... .......... .......... .......... 18%  308K 26s
  2350K .......... .......... .......... .......... .......... 19%  248K 26s
  2400K .......... .......... .......... .......... .......... 19%  314K 26s
  2450K .......... .......... .......... .......... .......... 20%  360K 26s
  2500K .......... .......... .......... .......... .......... 20%  345K 26s
  2550K .......... .......... .......... .......... .......... 20%  374K 26s
  2600K .......... .......... .......... .......... .......... 21%  457K 26s
  2650K .......... .......... .......... .......... .......... 21%  390K 26s
  2700K .......... .......... .......... .......... .......... 22%  452K 25s
  2750K .......... .......... .......... .......... .......... 22%  296K 25s
  2800K .......... .......... .......... .......... .......... 22%  499K 25s
  2850K .......... .......... .......... .......... .......... 23%  504K 25s
  2900K .......... .......... .......... .......... .......... 23%  391K 25s
  2950K .......... .......... .......... .......... .......... 24%  607K 25s
  3000K .......... .......... .......... .......... .......... 24%  653K 24s
  3050K .......... .......... .......... .......... .......... 24%  697K 24s
  3100K .......... .......... .......... .......... .......... 25%  691K 24s
  3150K .......... .......... .......... .......... .......... 25%  304K 24s
  3200K .......... .......... .......... .......... .......... 26%  599K 23s
  3250K .......... .......... .......... .......... .......... 26%  477K 23s
  3300K .......... .......... .......... .......... .......... 26%  434K 23s
  3350K .......... .......... .......... .......... .......... 27%  709K 23s
  3400K .......... .......... .......... .......... .......... 27%  583K 22s
  3450K .......... .......... .......... .......... .......... 28%  422K 22s
  3500K .......... .......... .......... .......... .......... 28%  348K 22s
  3550K .......... .......... .......... .......... .......... 28%  305K 22s
  3600K .......... .......... .......... .......... .......... 29%  560K 22s
  3650K .......... .......... .......... .......... .......... 29%  432K 22s
  3700K .......... .......... .......... .......... .......... 30%  753K 22s
  3750K .......... .......... .......... .......... .......... 30%  373K 22s
  3800K .......... .......... .......... .......... .......... 31%  390K 21s
  3850K .......... .......... .......... .......... .......... 31%  283K 21s
  3900K .......... .......... .......... .......... .......... 31%  271K 21s
  3950K .......... .......... .......... .......... .......... 32%  148K 22s
  4000K .......... .......... .......... .......... .......... 32%  129K 22s
  4050K .......... .......... .......... .......... .......... 33%  224K 22s
  4100K .......... .......... .......... .......... .......... 33%  322K 22s
  4150K .......... .......... .......... .......... .......... 33%  332K 22s
  4200K .......... .......... .......... .......... .......... 34%  316K 22s
  4250K .......... .......... .......... .......... .......... 34%  429K 22s
  4300K .......... .......... .......... .......... .......... 35%  289K 22s
  4350K .......... .......... .......... .......... .......... 35%  291K 22s
  4400K .......... .......... .......... .......... .......... 35%  319K 21s
  4450K .......... .......... .......... .......... .......... 36%  305K 21s
  4500K .......... .......... .......... .......... .......... 36%  246K 21s
  4550K .......... .......... .......... .......... .......... 37%  620K 21s
  4600K .......... .......... .......... .......... .......... 37%  518K 21s
  4650K .......... .......... .......... .......... .......... 37%  290K 21s
  4700K .......... .......... .......... .......... .......... 38%  395K 21s
  4750K .......... .......... .......... .......... .......... 38%  149K 21s
  4800K .......... .......... .......... .......... .......... 39%  283K 21s
  4850K .......... .......... .......... .......... .......... 39%  166K 21s
  4900K .......... .......... .......... .......... .......... 39% 3.67K 41s
  4950K .......... .......... .......... .......... .......... 40% 3.40M 40s
  5000K .......... .......... .......... .......... .......... 40% 7.03M 40s
  5050K .......... .......... .......... .......... .......... 41% 7.10M 39s
  5100K .......... .......... .......... .......... .......... 41% 7.24M 39s
  5150K .......... .......... .......... .......... .......... 41% 5.82M 38s
  5200K .......... .......... .......... .......... .......... 42% 7.15M 37s
  5250K .......... .......... .......... .......... .......... 42% 7.25M 37s
  5300K .......... .......... .......... .......... .......... 43% 7.23M 36s
  5350K .......... .......... .......... .......... .......... 43% 8.91M 36s
  5400K .......... .......... .......... .......... .......... 43% 13.6M 35s
  5450K .......... .......... .......... .......... .......... 44% 12.9M 34s
  5500K .......... .......... .......... .......... .......... 44% 13.6M 34s
  5550K .......... .......... .......... .......... .......... 45% 14.3M 33s
  5600K .......... .......... .......... .......... .......... 45% 12.4M 33s
  5650K .......... .......... .......... .......... .......... 45% 15.6M 32s
  5700K .......... .......... .......... .......... .......... 46% 16.1M 32s
  5750K .......... .......... .......... .......... .......... 46% 14.4M 31s
  5800K .......... .......... .......... .......... .......... 47% 15.6M 31s
  5850K .......... .......... .......... .......... .......... 47% 15.7M 30s
  5900K .......... .......... .......... .......... .......... 47% 15.8M 30s
  5950K .......... .......... .......... .......... .......... 48%  558K 29s
  6000K .......... .......... .......... .......... .......... 48%  193K 29s
  6050K .......... .......... .......... .......... .......... 49% 1.09M 29s
  6100K .......... .......... .......... .......... .......... 49%  615K 28s
  6150K .......... .......... .......... .......... .......... 49%  705K 28s
  6200K .......... .......... .......... .......... .......... 50%  815K 28s
  6250K .......... .......... .......... .......... .......... 50%  913K 27s
  6300K .......... .......... .......... .......... .......... 51%  801K 27s
  6350K .......... .......... .......... .......... .......... 51%  600K 27s
  6400K .......... .......... .......... .......... .......... 51%  349K 26s
  6450K .......... .......... .......... .......... .......... 52%  921K 26s
  6500K .......... .......... .......... .......... .......... 52% 1.01M 25s
  6550K .......... .......... .......... .......... .......... 53% 1.11M 25s
  6600K .......... .......... .......... .......... .......... 53%  778K 25s
  6650K .......... .......... .......... .......... .......... 53% 1.41M 24s
  6700K .......... .......... .......... .......... .......... 54%  851K 24s
  6750K .......... .......... .......... .......... .......... 54%  500K 24s
  6800K .......... .......... .......... .......... .......... 55%  547K 23s
  6850K .......... .......... .......... .......... .......... 55%  317K 23s
  6900K .......... .......... .......... .......... .......... 55%  413K 23s
  6950K .......... .......... .......... .......... .......... 56%  506K 23s
  7000K .......... .......... .......... .......... .......... 56%  449K 22s
  7050K .......... .......... .......... .......... .......... 57%  681K 22s
  7100K .......... .......... .......... .......... .......... 57%  607K 22s
  7150K .......... .......... .......... .......... .......... 57%  336K 21s
  7200K .......... .......... .......... .......... .......... 58%  672K 21s
  7250K .......... .......... .......... .......... .......... 58%  379K 21s
  7300K .......... .......... .......... .......... .......... 59%  365K 21s
  7350K .......... .......... .......... .......... .......... 59%  249K 20s
  7400K .......... .......... .......... .......... .......... 59%  316K 20s
  7450K .......... .......... .......... .......... .......... 60%  234K 20s
  7500K .......... .......... .......... .......... .......... 60%  605K 20s
  7550K .......... .......... .......... .......... .......... 61%  417K 19s
  7600K .......... .......... .......... .......... .......... 61%  567K 19s
  7650K .......... .......... .......... .......... .......... 62%  591K 19s
  7700K .......... .......... .......... .......... .......... 62%  434K 19s
  7750K .......... .......... .......... .......... .......... 62%  881K 18s
  7800K .......... .......... .......... .......... .......... 63%  605K 18s
  7850K .......... .......... .......... .......... .......... 63%  575K 18s
  7900K .......... .......... .......... .......... .......... 64%  922K 18s
  7950K .......... .......... .......... .......... .......... 64%  639K 17s
  8000K .......... .......... .......... .......... .......... 64%  867K 17s
  8050K .......... .......... .......... .......... .......... 65% 1.00M 17s
  8100K .......... .......... .......... .......... .......... 65%  985K 17s
  8150K .......... .......... .......... .......... .......... 66%  554K 16s
  8200K .......... .......... .......... .......... .......... 66%  469K 16s
  8250K .......... .......... .......... .......... .......... 66%  822K 16s
  8300K .......... .......... .......... .......... .......... 67% 1.27M 16s
  8350K .......... .......... .......... .......... .......... 67%  434K 15s
  8400K .......... .......... .......... .......... .......... 68%  539K 15s
  8450K .......... .......... .......... .......... .......... 68%  554K 15s
  8500K .......... .......... .......... .......... .......... 68%  606K 15s
  8550K .......... .......... .......... .......... .......... 69%  478K 14s
  8600K .......... .......... .......... .......... .......... 69%  882K 14s
  8650K .......... .......... .......... .......... .......... 70%  398K 14s
  8700K .......... .......... .......... .......... .......... 70%  670K 14s
  8750K .......... .......... .......... .......... .......... 70%  579K 13s
  8800K .......... .......... .......... .......... .......... 71% 1.02M 13s
  8850K .......... .......... .......... .......... .......... 71%  576K 13s
  8900K .......... .......... .......... .......... .......... 72%  489K 13s
  8950K .......... .......... .......... .......... .......... 72%  640K 13s
  9000K .......... .......... .......... .......... .......... 72%  409K 12s
  9050K .......... .......... .......... .......... .......... 73%  129K 12s
  9100K .......... .......... .......... .......... .......... 73%  153K 12s
  9150K .......... .......... .......... .......... .......... 74%  242K 12s
  9200K .......... .......... .......... .......... .......... 74%  251K 12s
  9250K .......... .......... .......... .......... .......... 74%  272K 12s
  9300K .......... .......... .......... .......... .......... 75%  584K 11s
  9350K .......... .......... .......... .......... .......... 75%  408K 11s
  9400K .......... .......... .......... .......... .......... 76%  405K 11s
  9450K .......... .......... .......... .......... .......... 76%  431K 11s
  9500K .......... .......... .......... .......... .......... 76%  318K 11s
  9550K .......... .......... .......... .......... .......... 77%  247K 10s
  9600K .......... .......... .......... .......... .......... 77%  337K 10s
  9650K .......... .......... .......... .......... .......... 78%  524K 10s
  9700K .......... .......... .......... .......... .......... 78%  426K 10s
  9750K .......... .......... .......... .......... .......... 78%  413K 10s
  9800K .......... .......... .......... .......... .......... 79%  380K 9s
  9850K .......... .......... .......... .......... .......... 79%  281K 9s
  9900K .......... .......... .......... .......... .......... 80%  479K 9s
  9950K .......... .......... .......... .......... .......... 80%  472K 9s
 10000K .......... .......... .......... .......... .......... 80%  402K 9s
 10050K .......... .......... .......... .......... .......... 81%  434K 8s
 10100K .......... .......... .......... .......... .......... 81%  550K 8s
 10150K .......... .......... .......... .......... .......... 82%  605K 8s
 10200K .......... .......... .......... .......... .......... 82%  529K 8s
 10250K .......... .......... .......... .......... .......... 82%  650K 8s
 10300K .......... .......... .......... .......... .......... 83%  347K 7s
 10350K .......... .......... .......... .......... .......... 83%  469K 7s
 10400K .......... .......... .......... .......... .......... 84%  392K 7s
 10450K .......... .......... .......... .......... .......... 84%  456K 7s
 10500K .......... .......... .......... .......... .......... 84%  755K 7s
 10550K .......... .......... .......... .......... .......... 85%  321K 6s
 10600K .......... .......... .......... .......... .......... 85%  632K 6s
 10650K .......... .......... .......... .......... .......... 86%  386K 6s
 10700K .......... .......... .......... .......... .......... 86%  404K 6s
 10750K .......... .......... .......... .......... .......... 86%  186K 6s
 10800K .......... .......... .......... .......... .......... 87%  216K 6s
 10850K .......... .......... .......... .......... .......... 87%  360K 5s
 10900K .......... .......... .......... .......... .......... 88%  407K 5s
 10950K .......... .......... .......... .......... .......... 88%  387K 5s
 11000K .......... .......... .......... .......... .......... 88%  526K 5s
 11050K .......... .......... .......... .......... .......... 89%  369K 5s
 11100K .......... .......... .......... .......... .......... 89%  512K 4s
 11150K .......... .......... .......... .......... .......... 90%  171K 4s
 11200K .......... .......... .......... .......... .......... 90%  281K 4s
 11250K .......... .......... .......... .......... .......... 90%  332K 4s
 11300K .......... .......... .......... .......... .......... 91%  302K 4s
 11350K .......... .......... .......... .......... .......... 91%  733K 4s
 11400K .......... .......... .......... .......... .......... 92%  329K 3s
 11450K .......... .......... .......... .......... .......... 92%  612K 3s
 11500K .......... .......... .......... .......... .......... 93%  456K 3s
 11550K .......... .......... .......... .......... .......... 93%  349K 3s
 11600K .......... .......... .......... .......... .......... 93%  677K 3s
 11650K .......... .......... .......... .......... .......... 94% 1007K 2s
 11700K .......... .......... .......... .......... .......... 94%  699K 2s
 11750K .......... .......... .......... .......... .......... 95%  570K 2s
 11800K .......... .......... .......... .......... .......... 95%  391K 2s
 11850K .......... .......... .......... .......... .......... 95%  377K 2s
 11900K .......... .......... .......... .......... .......... 96%  542K 2s
 11950K .......... .......... .......... .......... .......... 96%  147K 1s
 12000K .......... .......... .......... .......... .......... 97%  245K 1s
 12050K .......... .......... .......... .......... .......... 97%  394K 1s
 12100K .......... .......... .......... .......... .......... 97%  292K 1s
 12150K .......... .......... .......... .......... .......... 98%  407K 1s
 12200K .......... .......... .......... .......... .......... 98%  633K 1s
 12250K .......... .......... .......... .......... .......... 99%  400K 0s
 12300K .......... .......... .......... .......... .......... 99%  502K 0s
 12350K .......... .......... .......... .......... .......... 99%  545K 0s
 12400K .......... ........                                   100%  469K=42s

2019-02-01 17:46:01 (293 KB/s) - ‘node.tar.gz’ saved [12716600/12716600]

You have a global gyp installed. Setting up VirtualEnv without global pakages
--2019-02-01 17:46:17--  https://pypi.python.org/packages/source/v/virtualenv/virtualenv-12.0.7.tar.gz
Resolving pypi.python.org (pypi.python.org)... 151.101.112.223, 2a04:4e42:1b::223
Connecting to pypi.python.org (pypi.python.org)|151.101.112.223|:443... connected.
HTTP request sent, awaiting response... 301 Redirect to Primary Domain
Location: https://pypi.org/packages/source/v/virtualenv/virtualenv-12.0.7.tar.gz [following]
--2019-02-01 17:46:18--  https://pypi.org/packages/source/v/virtualenv/virtualenv-12.0.7.tar.gz
Resolving pypi.org (pypi.org)... 151.101.0.223, 151.101.64.223, 151.101.192.223, ...
Connecting to pypi.org (pypi.org)|151.101.0.223|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://files.pythonhosted.org/packages/source/v/virtualenv/virtualenv-12.0.7.tar.gz [following]
--2019-02-01 17:46:18--  https://files.pythonhosted.org/packages/source/v/virtualenv/virtualenv-12.0.7.tar.gz
Resolving files.pythonhosted.org (files.pythonhosted.org)... 151.101.113.63, 2a04:4e42:1b::319
Connecting to files.pythonhosted.org (files.pythonhosted.org)|151.101.113.63|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://files.pythonhosted.org/packages/14/a4/3145f84afd46a3a74717d97f0c58a7a6442876e7712b209c5070cf34a3be/virtualenv-12.0.7.tar.gz [following]
--2019-02-01 17:46:19--  https://files.pythonhosted.org/packages/14/a4/3145f84afd46a3a74717d97f0c58a7a6442876e7712b209c5070cf34a3be/virtualenv-12.0.7.tar.gz
Reusing existing connection to files.pythonhosted.org:443.
HTTP request sent, awaiting response... 200 OK
Length: 1829757 (1.7M) [application/octet-stream]
Saving to: ‘virtualenv-12.0.7.tar.gz’

     0K .......... .......... .......... .......... ..........  2%  451K 4s
    50K .......... .......... .......... .......... ..........  5%  459K 4s
   100K .......... .......... .......... .......... ..........  8%  703K 3s
   150K .......... .......... .......... .......... .......... 11%  938K 3s
   200K .......... .......... .......... .......... .......... 13%  645K 3s
   250K .......... .......... .......... .......... .......... 16%  748K 2s
   300K .......... .......... .......... .......... .......... 19% 1022K 2s
   350K .......... .......... .......... .......... .......... 22%  242K 3s
   400K .......... .......... .......... .......... .......... 25%  977K 2s
   450K .......... .......... .......... .......... .......... 27%  371K 2s
   500K .......... .......... .......... .......... .......... 30%  851K 2s
   550K .......... .......... .......... .......... .......... 33%  458K 2s
   600K .......... .......... .......... .......... .......... 36%  278K 2s
   650K .......... .......... .......... .......... .......... 39%  620K 2s
   700K .......... .......... .......... .......... .......... 41%  356K 2s
   750K .......... .......... .......... .......... .......... 44%  606K 2s
   800K .......... .......... .......... .......... .......... 47%  538K 2s
   850K .......... .......... .......... .......... .......... 50%  821K 2s
   900K .......... .......... .......... .......... .......... 53%  869K 2s
   950K .......... .......... .......... .......... .......... 55%  460K 1s
  1000K .......... .......... .......... .......... .......... 58%  376K 1s
  1050K .......... .......... .......... .......... .......... 61% 2.42M 1s
  1100K .......... .......... .......... .......... .......... 64%  331K 1s
  1150K .......... .......... .......... .......... .......... 67%  934K 1s
  1200K .......... .......... .......... .......... .......... 69%  138K 1s
  1250K .......... .......... .......... .......... .......... 72% 1.34M 1s
  1300K .......... .......... .......... .......... .......... 75%  370K 1s
  1350K .......... .......... .......... .......... .......... 78%  355K 1s
  1400K .......... .......... .......... .......... .......... 81%  734K 1s
  1450K .......... .......... .......... .......... .......... 83%  517K 1s
  1500K .......... .......... .......... .......... .......... 86%  309K 0s
  1550K .......... .......... .......... .......... .......... 89%  328K 0s
  1600K .......... .......... .......... .......... .......... 92% 1.27M 0s
  1650K .......... .......... .......... .......... .......... 95%  378K 0s
  1700K .......... .......... .......... .......... .......... 97% 2.44M 0s
  1750K .......... .......... .......... ......               100% 1.25M=3.6s

2019-02-01 17:46:23 (492 KB/s) - ‘virtualenv-12.0.7.tar.gz’ saved [1829757/1829757]

New python executable in /home/homeassistant/.c9/python/bin/python2.7
Also creating executable in /home/homeassistant/.c9/python/bin/python
Installing setuptools, pip...done.

real    2m18.479s
user    0m14.886s
sys     0m2.409s
:Installing TMUX
:Downloading tmux source code
:N.B: This will take a while. To speed this up install tmux 2.2 manually on your machine and restart this process.
:Downloading Libevent...
--2019-02-01 17:47:30--  https://raw.githubusercontent.com/c9/install/master/packages/tmux/libevent-2.1.8-stable.tar.gz
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.112.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.112.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1026485 (1002K) [application/octet-stream]
Saving to: ‘libevent-2.1.8-stable.tar.gz’

     0K .......... .......... .......... .......... ..........  4%  305K 3s
    50K .......... .......... .......... .......... ..........  9%  711K 2s
   100K .......... .......... .......... .......... .......... 14% 1.91M 1s
   150K .......... .......... .......... .......... .......... 19%  325K 2s
   200K .......... .......... .......... .......... .......... 24%  858K 1s
   250K .......... .......... .......... .......... .......... 29%  525K 1s
   300K .......... .......... .......... .......... .......... 34%  758K 1s
   350K .......... .......... .......... .......... .......... 39%  459K 1s
   400K .......... .......... .......... .......... .......... 44%  908K 1s
   450K .......... .......... .......... .......... .......... 49% 1.09M 1s
   500K .......... .......... .......... .......... .......... 54%  481K 1s
   550K .......... .......... .......... .......... .......... 59% 2.60M 1s
   600K .......... .......... .......... .......... .......... 64%  596K 1s
   650K .......... .......... .......... .......... .......... 69% 1.54M 0s
   700K .......... .......... .......... .......... .......... 74% 1.87M 0s
   750K .......... .......... .......... .......... .......... 79%  504K 0s
   800K .......... .......... .......... .......... .......... 84% 1.30M 0s
   850K .......... .......... .......... .......... .......... 89%  819K 0s
   900K .......... .......... .......... .......... .......... 94%  858K 0s
   950K .......... .......... .......... .......... .......... 99%  409K 0s
  1000K ..                                                    100% 4629G=1.5s

2019-02-01 17:47:32 (675 KB/s) - ‘libevent-2.1.8-stable.tar.gz’ saved [1026485/1026485]

:Downloading Ncurses...
File `ncurses-6.0.tar.gz' already there; not retrieving.

real    0m3.113s
user    0m0.227s
sys     0m0.069s
Creating workspace for Cloud9.
Create default config.
{"projecttree": {"@showhidden": false,"@hiddenFilePattern": ".*,.n*,*c9*,.b*,.p*,.w*,*.db"}}
Symlinking /home/homeassistant/.homeassistant to the workspace.
Copying Cloud9 service file...
Enabling Cloud9 service...
Created symlink /etc/systemd/system/multi-user.target.wants/cloud9@homeassistant.service → /etc/systemd/system/cloud9@homeassistant.service.
Starting Cloud9 service...
Checking the installation...

Installation done.
Your Cloud9 IDE is now avaiable at http://192.168.1.21:8181
</pre>

I rebooted the Raspberry but service still fails:

<pre>
 sudo systemctl status cloud9@homeassistant.service
</pre>

<pre>
● cloud9@homeassistant.service - Cloud9
   Loaded: loaded (/etc/systemd/system/cloud9@homeassistant.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2019-02-01 17:49:09 CET; 4min 3s ago
  Process: 500 ExecStart=/opt/c9sdk/server.js -l 0.0.0.0 -a : -w /home/homeassistant/c9workspace (code=exited, status=1/FAILURE)
 Main PID: 500 (code=exited, status=1/FAILURE)

Feb 01 17:49:05 Raspy systemd[1]: Started Cloud9.
Feb 01 17:49:09 Raspy systemd[1]: cloud9@homeassistant.service: Main process exited, code=exited, status=1/FAILURE
Feb 01 17:49:09 Raspy systemd[1]: cloud9@homeassistant.service: Unit entered failed state.
Feb 01 17:49:09 Raspy systemd[1]: cloud9@homeassistant.service: Failed with result 'exit-code'.
</pre>

So I tried to run:

<pre>
/opt/c9sdk/server.js -l 0.0.0.0 -a : -w /home/homeassistant/c9workspace
</pre>

And this is the output:

<pre>
/opt/c9sdk/node_modules/c9/setup_paths.js:34
        throw e;
        ^

Error: Cannot find module 'optimist'
    at Module._resolveFilename (module.js:547:15)
    at Function.modules._resolveFilename (/opt/c9sdk/node_modules/c9/setup_paths.js:27:16)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at main (/opt/c9sdk/server.js:74:20)
    at Object.<anonymous> (/opt/c9sdk/server.js:55:5)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
</pre>

Don't know how to help you anymore

--- a/package/opt/hassbian/suites/cloud9.sh
+++ b/package/opt/hassbian/suites/cloud9.sh
@@ -24,6 +24,9 @@ echo "Creating installation directory..."
 mkdir /opt/c9sdk
 chown homeassistant:homeassistant /opt/c9sdk
 
+echo "Installing npm"
+apt isntall -y npm
+
 echo "Changing to the homeassistant user"
 sudo -u homeassistant -H /bin/bash << EOF
   printf "Downloading and installing Cloud9 SDK...\\n"
@@ -36,6 +39,9 @@ sudo -u homeassistant -H /bin/bash << EOF
   echo "Symlinking /home/homeassistant/.homeassistant to the workspace."
   ln -s /home/homeassistant/.homeassistant/ /home/homeassistant/c9workspace/homeassistant
 EOF
+
+cd /opt/c9sdk || exit |
+npm install
 
 echo "Copying Cloud9 service file..."
 cp /opt/hassbian/suites/files/cloud9.service /etc/systemd/system/cloud9@homeassistant.service

--- a/package/opt/hassbian/suites/cloud9.sh
+++ b/package/opt/hassbian/suites/cloud9.sh
@@ -24,9 +24,6 @@ echo "Creating installation directory..."
 mkdir /opt/c9sdk
 chown homeassistant:homeassistant /opt/c9sdk
 
-echo "Installing npm"
-apt install -y npm
-
 echo "Changing to the homeassistant user"
 sudo -u homeassistant -H /bin/bash << EOF
   printf "Downloading and installing Cloud9 SDK...\\n"
@@ -40,9 +37,6 @@ sudo -u homeassistant -H /bin/bash << EOF
   ln -s /home/homeassistant/.homeassistant/ /home/homeassistant/c9workspace/homeassistant
 EOF
 
-cd /opt/c9sdk || exit 1
-npm install
-
 echo "Copying Cloud9 service file..."
 cp /opt/hassbian/suites/files/cloud9.service /etc/systemd/system/cloud9@homeassistant.service
 
@@ -54,9 +48,23 @@ echo "Starting Cloud9 service..."
 systemctl start cloud9@homeassistant.service
 
 echo "Checking the installation..."
-ip_address=$(ifconfig | grep "inet.*broadcast" | grep -v 0.0.0.0 | awk '{print $2}')
+sleep 15
 validation=$(pgrep -f cloud9)
 if [ ! -z "${validation}" ]; then
+  echo "Using fallback installation."
+  echo "Installing npm"
+  apt install -y npm
+
+  cd /opt/c9sdk || exit 1
+  npm install
+
+  echo "Checking the installation..."
+  sleep 15
+fi
+
+validation=$(pgrep -f cloud9)
+if [ -z "${validation}" ]; then
+  ip_address=$(ifconfig | grep "inet.*broadcast" | grep -v 0.0.0.0 | awk '{print $2}')
   echo
   echo -e "\\e[32mInstallation done.\\e[0m"
   echo "Your Cloud9 IDE is now avaiable at http://$ip_address:8181"
@@ -83,6 +91,7 @@ printf "Starting Cloud9 service...\\n"
 systemctl start cloud9@homeassistant.service
 
 echo "Checking the installation..."
+sleep 15
 validation=$(pgrep -f cloud9)
 if [ ! -z "${validation}" ]; then
   echo

--- a/package/opt/hassbian/suites/cloud9.sh
+++ b/package/opt/hassbian/suites/cloud9.sh
@@ -40,7 +40,7 @@ sudo -u homeassistant -H /bin/bash << EOF
   ln -s /home/homeassistant/.homeassistant/ /home/homeassistant/c9workspace/homeassistant
 EOF
 
-cd /opt/c9sdk || exit |
+cd /opt/c9sdk || exit 1
 npm install
 
 echo "Copying Cloud9 service file..."

--- a/package/opt/hassbian/suites/cloud9.sh
+++ b/package/opt/hassbian/suites/cloud9.sh
@@ -25,7 +25,7 @@ mkdir /opt/c9sdk
 chown homeassistant:homeassistant /opt/c9sdk
 
 echo "Installing npm"
-apt isntall -y npm
+apt install -y npm
 
 echo "Changing to the homeassistant user"
 sudo -u homeassistant -H /bin/bash << EOF

